k8s/plugin/src/resources/mod.rs

```diff
@@ -35,14 +35,19 @@ pub struct CliArgs {
     #[clap(skip)]
     pub namespace: String,
 
+    /// Kubernetes context in the kubeconfig file.
+    #[clap(skip)]
+    pub context: Option<String>,
+
     #[clap(flatten)]
     pub cli_args: plugin::CliArgs,
 }
 
 impl CliArgs {
     /// The kube client, with the correct install namespace.
     pub async fn client(&self) -> anyhow::Result<kube::Client> {
-        let mut config = kube_proxy::config_from_kubeconfig(self.kubeconfig.clone()).await?;
+        let opts = kube_proxy::kubeconfigoptions_from_context(self.context.clone());
```

Comment:
```suggestion
        let opts = kube_proxy::kube_configoptions_from_context(self.context.clone());
```

Continuing from the control-plane review

Reply:
Changed as suggested